docarray/index/backends/in_memory.py

```diff
@@ -0,0 +1,304 @@
+            )
+
+        ann_docs = DocList.__class_getitem__(cast(Type[BaseDoc], self._schema))([])
+        filter_conditions = []
+        doc_to_score: Dict[BaseDoc, Any] = {}
+        for op, op_kwargs in query:
+            if op == 'find':
+                docs, scores = self.find(**op_kwargs)
+                ann_docs.extend(docs)
+                doc_to_score.update(zip(docs.__getattribute__('id'), scores))
+            elif op == 'filter':
+                filter_conditions.append(op_kwargs['filter_query'])
+
+        self._logger.debug(f'Executing query {query}')
+        docs_filtered = ann_docs
+        for cond in filter_conditions:
+            docs_cls = DocList.__class_getitem__(cast(Type[BaseDoc], self._schema))
+            docs_filtered = docs_cls(filter_docs(docs_filtered, cond))
+
+        self._logger.debug(f'{len(docs_filtered)} results found')
+        docs_and_scores = zip(
+            docs_filtered, (doc_to_score[doc.id] for doc in docs_filtered)
+        )
+        docs_sorted = sorted(docs_and_scores, key=lambda x: x[1])
+        out_docs, out_scores = zip(*docs_sorted)
+        return FindResult(documents=out_docs, scores=out_scores)
+
+    def find(
+        self,
+        query: Union[AnyTensor, BaseDoc],
+        search_field: str = '',
+        limit: int = 10,
+        **kwargs,
+    ) -> FindResult:
+        """Find documents in the index using nearest neighbor search.
+
+        :param query: query vector for KNN/ANN search.
+            Can be either a tensor-like (np.array, torch.Tensor, etc.)
+            with a single axis, or a Document
+        :param search_field: name of the field to search on.
+            Documents in the index are retrieved based on this similarity
+            of this field to the query.
+        :param limit: maximum number of documents to return
+        :return: a named tuple containing `documents` and `scores`
+        """
+        self._logger.debug(f'Executing `find` for search field {search_field}')
+        self._validate_search_field(search_field)
+        config = self._column_infos[search_field].config
+
+        docs, scores = find(
+            index=self._docs,
+            query=query,
+            search_field=search_field,
+            limit=limit,
+            metric=config['space'],
+        )
+        docs_with_schema = DocList.__class_getitem__(cast(Type[BaseDoc], self._schema))(
+            docs
+        )
+        return FindResult(documents=docs_with_schema, scores=scores)
+
+    def _find(
+        self, query: np.ndarray, limit: int, search_field: str = ''
+    ) -> _FindResult:
+        raise NotImplementedError
+
+    def find_batched(
+        self,
+        queries: Union[AnyTensor, DocList],
+        search_field: str = '',
+        limit: int = 10,
+        **kwargs,
+    ) -> FindResultBatched:
+        """Find documents in the index using nearest neighbor search.
```

Comment:
```suggestion
        """Find Documents in the index using nearest-neighbor search.
```
Ensure consistent hyphenation throughout